build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java

```diff
@@ -346,21 +346,27 @@ public MBSWizardHandler(IBuildPropertyValue val, Composite p, IWizard w) {
 		propertyId = val.getId();
 		setWizard(w);
 	}
+
 	private void setWizard(IWizard w) {
 		if (w != null) {
-			if (w.getStartingPage() instanceof IWizardItemsListListener)
-				listener = (IWizardItemsListListener)w.getStartingPage();
 			wizard = w;
-			startingPage = w.getStartingPage();
+			for (IWizardPage page : w.getPages()) {
+				if (page instanceof WizardNewProjectCreationPage) {
+					newProjectCreationPage = page;
+					break;
+				}
+			}
+			if (newProjectCreationPage instanceof IWizardItemsListListener)
+				listener = (IWizardItemsListListener) newProjectCreationPage;
 		}
 	}
 
-	protected IWizardPage getStartingPage(){
-		return startingPage;
+	protected IWizardPage getNewProjectCreationPage(){
```

Comment:
Any non-private method in an non-internal class is generally considered API, so we cannot rename methods. If suitable, a new method can be created and the existing one deprecated though.